<?xml version="1.0"?>
<robot name="scara" xmlns:xacro="http://ros.org/wiki/xacro">

  <material name="red">
    <color rgba="0.8 0.1 0.1 1"/>
  </material>
  
  <material name="blue">
    <color rgba="0.1 0.1 0.8 1"/>
  </material>
  
  <material name="green">
    <color rgba="0.1 0.8 0.1 1"/>
  </material>
  
  <material name="orange">
    <color rgba="1 0.5 0 1"/>
  </material>
  
  <material name="gray">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>

  <link name="base_link">
    <visual>
      <geometry>
        <cylinder radius="0.05" length="0.04"/>
      </geometry>
      <origin xyz="0 0 0.02" rpy="0 0 0"/>
      <material name="gray"/>
    </visual>
  </link>

  <link name="link1">
    <visual>
      <geometry>
        <box size="0.1445 0.02 0.015"/>
      </geometry>
      <origin xyz="0.07225 0 0" rpy="0 0 0"/>
      <material name="red"/>
    </visual>
    <inertial>
        <mass value="0.105"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <inertia ixx="0.0025" ixy="0" ixz="0" iyy="0.0034" iyz="0" izz="0.0034"/>
    </inertial>

    <collision>
      <geometry>
          <box size="0.1445 0.02 0.015"/>
      </geometry>
      <origin xyz="0.07225 0 0" rpy="0 0 0"/>
    </collision>
  </link>

  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0.04" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit lower="-1.08" upper="1.08" effort="0.05" velocity="2"/>
  </joint>

  <link name="link2">
    <visual>
      <geometry>
        <box size="0.1445 0.018 0.012"/>
      </geometry>
      <origin xyz="0.07225 0 0" rpy="0 0 0"/>
      <material name="blue"/>
    </visual>
    <inertial>
          <mass value="0.092"/>
          <origin xyz="0 0 0" rpy="0 0 0"/>
          <inertia ixx="0.0019" ixy="0" ixz="0" iyy="0.0026" iyz="0" izz="0.0026"/>
      </inertial>

      <collision>
        <geometry>
          <box size="0.1445 0.018 0.012"/>
        </geometry>
        <origin xyz="0.07225 0 0" rpy="0 0 0"/>
      </collision>
  </link>

  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0.1445 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit lower="-2.26" upper="2.26" effort="0.02" velocity="2"/>
  </joint>

  <link name="link3">
    <visual>
      <geometry>
        <cylinder radius="0.004" length="0.04"/>
      </geometry>
      <origin xyz="0 0 -0.02" rpy="0 0 0"/>
      <material name="green"/>
    </visual>
    <inertial>
        <mass value="0.008"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <inertia ixx="0.0003" ixy="0" ixz="0" iyy="0.0003" iyz="0" izz="0.0001"/>
    </inertial>

    <collision>
      <geometry>
        <cylinder radius="0.004" length="0.04"/>
      </geometry>
      <origin xyz="0 0 -0.02" rpy="0 0 0"/>
    </collision>
  </link>

  <joint name="joint3" type="prismatic">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="0.1445 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 -1"/>
    <limit lower="-0.01" upper="0.0" effort="0.08" velocity="0.015"/>
  </joint>

</robot>
